cmds/basename/basename.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2012-2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Basename return name with leading path information removed.
+//
+// Synopsis:
+//     basename NAME [SUFFIX]
+//
+// Options:
+//     -s: optional flag for removing suffix
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+)
+
+var (
+	suffix = flag.String("s", "", "Strip prefix from file")
```

Comment:
To simplify things, it would be alright not to have this flag. There are two ways to invoke basename:

    basename y/src/hello.c    # => hello.c
    basename y/src/hello.c .c # => hello

This is posix and compatible with gnu coreutils.

See: http://pubs.opengroup.org/onlinepubs/009695299/utilities/basename.html

Reply:
This certainly makes it much easier 👍